### DISCUSSION TITLE
SHA256 mismatch with R cask

### DISCUSSION THREAD

**Discussion by mrmorgan17** [Homebrew/discussions#3208] [Casks] [answered]
Trying to update R to version 4.2.0 from 4.1.3 through Homebrew
```
brew cleanup && brew update

Already up-to-date.
r (4.1.3) != 4.2.0
==> Casks with 'auto_updates true' or 'version :latest' will not be upgraded; pass `--greedy` to upgrade them.
==> Upgrading 1 outdated package:
r 4.1.3 -> 4.2.0
==> Upgrading r
==> Caveats
Cask r installs files under /usr/local. The presence of such
files can cause warnings when running `brew doctor`, which is considered
to be a bug in Homebrew Cask.

==> Downloading https://cloud.r-project.org/bin/macosx/base/R-4.2.0.pkg
######################################################################## 100.0%
==> Purging files for version 4.2.0 of Cask r
Error: r: SHA256 mismatch
Expected: 8d1f1f51d04bdf89974f6e3aea7bf178f6db79f3d41d175c2375fa7bcbfef19d
  Actual: efb0a98d407c4a4d621cf3f3c33d674b71071099577301b079f4d114b2fc2d00
    File: /Users/matthewmorgan/Library/Caches/Homebrew/downloads/1aa54bb5bbc0ae775426afcc69d3fb71cab535e72e0749d0aaf3364ea60bde9d--R-4.2.0.pkg
To retry an incomplete download, remove the file above.
```

I have removed the file multiple times using
```
rm Library/Caches/Homebrew/downloads/1aa54bb5bbc0ae775426afcc69d3fb71cab535e72e0749d0aaf3364ea60bde9d--R-4.2.0.pkg
```
I even went through Finder on my Mac and deleted it that way too

Although every time I run `brew cleanup && brew update` after  removing the file, I get the same SHA256 mismatch prompt telling me retry an incomplete download

Any suggestions?

MacOS Monterrey 12.3.1
Homebrew 3.4.7

**Comment by gromgit** [comment, marked as answer]
`brew update`, then try the install again. There was an update to the R cask that fixed the checksums.

**Reply by mrmorgan17** [reply]
Yeah, looks like it just takes a little longer on Homebrew’s end. It worked when I tried things again later this afternoon

**Reply by SMillerDev** [reply]
You're always welcome to contribute to fix these things faster. See https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match
